packages/muze-legend/src/legend/legend-helper.js

```diff
@@ -27,16 +27,144 @@ export const getScaleInfo = (scale) => {
+    for (let i = 1; i < domainForLegend.length; i++) {
+        nextTickValue = ((currentTickValue / pixelPerTick) + (currentTickValue / 2) + domainForLegend[i - 1]);
+        if (domainForLegend[i] < nextTickValue) {
+            domainForLegend.splice(i, 1);
+            i -= 1;
+        }
+        currentTickValue = (smartLabelCalc.getOriSize((domainForLegend[i]))[param]);
+    }
+    return domainForLegend;
+};
+
+/**
+ * function to recompute the Stops Array provided to prevent the Overlapping of values
+ * @param  {Array} domainForLegend - Stops Array
+ * @param  {Object} scaleParams - Scale Parameters
+ * @return {Array} - modified Stops Array
+ */
+export const getInterpolatedArrayData = (domainForLegend, scaleParams) => {
+    /* Checking if UpperBound of Domain is Floating or Not.
+     In case of floating constricting it to 2 decimals after point. */
+    let upperBound = domainForLegend[domainForLegend.length - 1];
+    if (!Number.isInteger(upperBound)) {
+        upperBound = ((upperBound).toFixed(2));
+    }
+    // Initializing Minimum Tick Difference Variable and checking if it's less than 1 or not
+    let minTickDiff = (domainForLegend[1] - domainForLegend[0]);
+    minTickDiff = minTickDiff < 1 ? 1 : minTickDiff;
+
+    // defining param for height/width selector
+    let measureParam;
+
+    // gradient Alignment
+    const { alignment } = scaleParams;
+
+    // scale Measurements (i.e MaxWidth and MaxHeight available)
+    const availableSpace = scaleParams.measures;
+
+    // getting minimum and Maximum values in domain
+    const [min, max] = [Math.min(...domainForLegend), Math.max(...domainForLegend)];
+
+    // getting SmartLabel Manager to calculate tick Params
+    const smartLabelCalc = scaleParams.smartLabel;
+
+    // getting minimum Tick size (i.e height and width)
+    const minimumTickSize = scaleParams.minTickDistance;
+
+    // getting domain upperbound dimensions
+    const { height: tickDimHeight, width: tickDimWidth } = smartLabelCalc.getOriSize((upperBound));
+
+    // required width to render legend
+    const requiredWidth = ((max - min) / Math.abs(minTickDiff)) * (tickDimWidth + (minimumTickSize.width));
+
+    // require height to render legend
+    const requiredHeight = ((max - min) / Math.abs(minTickDiff)) * tickDimHeight;
+
+    // required Height and width to render
+    const requiredMeasure = {
+        height: requiredHeight,
+        width: requiredWidth
+    };
+
+    // max available spaces to render
+    const availableMeasure = {
+        height: availableSpace.maxHeight,
+        width: availableSpace.maxWidth
+    };
+
+    // checking the alignment of legend
+    if (alignment === TOP || alignment === BOTTOM) {
+        measureParam = WIDTH;
+    } else {
+        measureParam = HEIGHT;
+    }
+
+    // calculating computed array
+    domainForLegend = getcomputedArray(measureParam, requiredMeasure, availableMeasure,
```

Comment:
Try to restrict the number of arguments to a function to 4

Reply:
constricted the number of arguments to 4